include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1471,30 +1487,25 @@ void solveRank1ProblemDist(CommSender&& row_comm, CommSender&& col_comm, const S
         // STEP 2c: reduce, then finalize computation with sign and square root (single-thread)
         if (thread_idx == 0) {
           // local reduction from all bulk workers
-          for (SizeType i = 0; i < m_subm_el_lc; ++i) {
+          for (SizeType i = 0; i < m_el_lc; ++i) {
             for (std::size_t tidx = 1; tidx < nthreads; ++tidx) {
               const T* w_partial = ws_cols[tidx]();
               w[i] *= w_partial[i];
             }
           }
 
           tt::sync_wait(ex::when_all(row_comm_chain(),
-                                     ex::just(MPI_PROD, common::make_data(w, m_subm_el_lc))) |
+                                     ex::just(MPI_PROD, common::make_data(w, m_el_lc))) |
                         transformMPI(all_reduce_in_place));
 
+          // TODO check all weights < 0 (!= 0 otherwise q elements are set to zero and then nomr = 0 => nan)
+
           T* weights = ws_cols[nthreads]();
-          for (SizeType i_subm_el_lc = 0; i_subm_el_lc < m_subm_el_lc; ++i_subm_el_lc) {
-            const SizeType i_subm_lc = i_subm_el_lc / dist.blockSize().rows();
-            const SizeType i_lc = ij_begin_lc.row() + i_subm_lc;
-            const SizeType i = dist.globalTileFromLocalTile<Coord::Row>(i_lc);
-            const SizeType i_subm = i - i_begin;
-            const SizeType i_subm_el =
-                i_subm * dist.blockSize().rows() + i_subm_el_lc % dist.blockSize().rows();
-
-            const auto* i2_perm = i2_tile_arr[0].get().ptr();
-            const SizeType ii_subm_el = i2_perm[i_subm_el];
-            weights[to_sizet(i_subm_el_lc)] =
-                std::copysign(std::sqrt(-w[i_subm_el_lc]), z_ptr[to_sizet(ii_subm_el)]);
+          // TODO this can be limited to k_lc
```

Comment:
self-reminder: check if this is true

Reply:
it is likely not worth changing the logic since it would have to go through the permutation to skip deflated positions.